File: examples/preview.py
from dataclasses import dataclass
from enum import Enum


class Light(Enum):
    GOLDEN = "golden"
    PLUM = "plum"


@dataclass(frozen=True)
class Horizon:
    name: str
    intensity: float = 0.84

    def describe(self) -> str:
        """Return a compact sunset description."""
        return f"{self.name}: {self.intensity:.0%} glow"


sunset = Horizon("Afterglow")
print(sunset.describe())

File: examples/preview.rs
#[derive(Debug)]
struct Horizon { name: &'static str, glow: u8 }

fn main() {
    // Lifetimes, macros, types, and member access.
    let sky = Horizon { name: "Afterglow", glow: 84 };
    if sky.glow > 0 { println!("{}: {:?}", sky.name, sky); }
}
